{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "datasets/dbpedia-url.txt\n",
    "...........................................................................................................................................................................................................................................................................................................................................Absent urls: 16\n",
    "Total urls: 331\n",
    "Hits total for all urls: 111042\n",
    "Avg. hits per url: 335.47\n",
    "\n",
    " datasets/n3-reuters-128-urls.txt\n",
    "...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Absent urls: 309\n",
    "Total urls: 879\n",
    "Hits total for all urls: 50506\n",
    "Avg. hits per url: 57.46\n",
    "\n",
    " datasets/kore50-urls.txt\n",
    "................................................................................................................................................Absent urls: 15\n",
    "Total urls: 144\n",
    "Hits total for all urls: 1525\n",
    "Avg. hits per url: 10.59"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from time import time\n",
    "for x in 3, 32, 10, 100:\n",
    "    tic = time()\n",
    "    make_queries(queries, parallel=x)\n",
    "    print(x, time()-tic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for dataset_fpath in glob(datasets_fpaths):\n",
    "    with open(dataset_fpath, \"r\") as in_f:\n",
    "        queries = []\n",
    "        for url in in_f:\n",
    "            url = db2wiki(url.strip())\n",
    "            query = 'origins:\"{}\"'.format(url)\n",
    "            queries.append(query)\n",
    "            \n",
    "        print(queries)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "russians = saved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/panchenko/kb2vec/datasets/n3-news-128.tsv\n",
      "# of contexts: 129\n",
      "# of targets: 622\n",
      "/home/panchenko/kb2vec/datasets/kore50.tsv\n",
      "# of contexts: 50\n",
      "# of targets: 128\n",
      "/home/panchenko/kb2vec/datasets/dbpedia.tsv\n",
      "# of contexts: 59\n",
      "# of targets: 265\n",
      "/home/panchenko/kb2vec/datasets/n3-reuters-128.tsv\n",
      "# of contexts: 128\n",
      "# of targets: 614\n",
      "/home/panchenko/kb2vec/datasets/ace2004.tsv\n",
      "# of contexts: 57\n",
      "# of targets: 246\n",
      "/home/panchenko/kb2vec/datasets/dzerczynski.tsv\n",
      "# of contexts: 363\n",
      "# of targets: 273\n",
      "datasets/dbpedia-url.txt\n",
      "datasets/n3-reuters-128-urls.txt\n",
      "datasets/kore50-urls.txt\n",
      "datasets/dbpedia-url.txt\n",
      "en.wikipedia.org/wiki/Anxiety error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Conspiracy_theory error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Consumer error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Internet error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Internet_privacy error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Lawsuit error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Marketing error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/User_(computing) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Worry error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Alarmism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Advertising error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Marketing error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Online_and_offline error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/User_(computing) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Web_developer error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Year error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/HTML5 error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Information_privacy error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Internet error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Internet error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Era error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/HTML error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Promise error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Source_code error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Web_browser error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Web_page error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/World_Wide_Web error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Year error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Computer_software error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Content_(media) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Email error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Multimedia error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Online_and_offline error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Restaurant error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Smartphone error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Uploading_and_downloading error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/User_(computing) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Billionaire error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Country error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Home error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mikhail_Prokhorov error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Moscow error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/New_Jersey_Nets error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ownership error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sunday error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mikhail_Prokhorov error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Country error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Greeting error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Home error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Russia error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sponsor_(commercial) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sportsperson error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Team error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Wealth error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mikhail_Prokhorov error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Basketball_court error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/CSKA_Universal_Sports_Hall error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Exhibition_game error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Game error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/New_Jersey_Nets error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Practice_(learning_method) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Russia error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sports_club error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sportsperson error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Year error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mikhail_Prokhorov error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Association_football error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Basketball error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Coach_(sport) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Future error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Hockey error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Interaction error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/New_Jersey_Nets error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Popularity error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Russia error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Russians error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sport error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sportsperson error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Fan_(person) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/New_Jersey_Nets error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Cannabis_(drug) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Family error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Middle_age error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Parent error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ritual error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Cannabis_(drug) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Illegal_drug_trade error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Illinois error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Parent error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Writer error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Year error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Parent error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Illegal_drug_trade error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Grounding_(punishment) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mother error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Medical_cannabis error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Parent error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Alzheimer’s_disease error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Cancer error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Father error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Heart_disease error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mother error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Nausea error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Orthostatic_hypotension error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Medical_cannabis error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Research error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/California_State_Route_1 error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Driving error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Los_Angeles error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Roof error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/San_Francisco error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Toyota_Prius error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Driving error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Person error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Steering_wheel error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Driverless_car error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Artificial_intelligence error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Automobile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Automobile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Computer_software error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Google error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Human error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Project error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Automobile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Human error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Human error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Navigation_system error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Steering_wheel error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Technician error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Lombard_Street_(San_Francisco) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/San_Francisco error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Street error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Accident error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Automobile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Engineer error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Google_driverless_car error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Rear-end_collision error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Traffic_light error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Canadian error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dinner error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Diplomatic_mission error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "en.wikipedia.org/wiki/Economy error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Economy error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Friday error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/International error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Presidency_of_Barack_Obama error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Problem error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/World error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Balance_of_trade error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Beef_tenderloin error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Canada error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Currency error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Europe error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Japan error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/People's_Republic_of_China error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Renminbi error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Scallop error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Timothy_Geithner error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/United_States_Secretary_of_the_Treasury error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/World error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Annual_Meetings_of_the_International_Monetary_Fund_and_the_World_Bank_Group error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Currency error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Box_office error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Art error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Art_exhibition error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Art_exhibition error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Autumn error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Chaos error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Classicism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/France error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Germany error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Italy error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Name error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Percentage error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Solomon_R._Guggenheim_Museum error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/History error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Cubism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Futurism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Modernism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Nightmare error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/World_War_I error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ancient_Greece error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Italian_Renaissance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Tradition error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Artist error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Conservatism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ideology error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Social_order error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Lil_Wayne error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Boredom error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Free_association_(psychology) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Idea error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Self-consciousness error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Aggression error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Album error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Album error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Idea error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Prison error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Rebirth_(Lil_Wayne_album) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Research error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Rikers_Island error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Scientific_method error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sentence_(law) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Experiment error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Lil_Wayne error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Album error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/I_Am_Not_a_Human_Being error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Month error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Song error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Studio_recording error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Criminal_charge error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gun_politics error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Prison error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Rebirth_(Lil_Wayne_album) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Scar error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Song error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Accordion error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Backpack error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ballet_tutu error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Man error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Audience error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Buster_Keaton error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Chair error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ladder error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/2006 error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Berlin error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Brick error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Choreography error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gothic_architecture error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/House_dance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Man error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Performance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Pumping_station error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sasha_Waltz error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Spree error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Berlin error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Boiler error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dance_troupe error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Hall error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Machine error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Man error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Performance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Pumping_station error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Room error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Berlin_Ostbahnhof error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Boulevard error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Berlin error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Mile error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Nightclub error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Unter_den_Linden error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Cappuccino error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Dance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Month error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Performance error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Ticket_(admission) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Blue error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Finland error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Floor_plan error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Green error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/House error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Leaf error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Leaf_shape error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Park error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Red error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Roof error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Turku error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Yellow error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Commuting error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Curve error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Drop_(liquid) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Floor error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Grand_Central_Terminal error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/New_York_City error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Runway_(fashion) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Shape error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Video error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Wall error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Window error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Port_of_Turku error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/House error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Landmark error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/September error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Turku error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Woodland error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Artist error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Artist error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Interior_design error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sculpture error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Finnish_sauna error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/2011 error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Culture error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Exhibition error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Festival error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Fiberglass error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Sauna error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Garlic error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/House error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Red error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Shape error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Turku error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Alice_Roosevelt_Longworth error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Hedonism error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gossip error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gossip error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Research error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Husband error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Presentation error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/September error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Social_psychology error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Emotion error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Fiction error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gossip error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Man error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Person error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Person error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Term_(time) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Undergraduate_education error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Woman error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Female error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gossip error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Gossip error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Questionnaire error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Self-esteem error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Social_support error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Term_(time) error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "en.wikipedia.org/wiki/Undergraduate_education error\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/panchenko/kb2vec/dataset_stats.py\", line 35, in generate_absent_datasets\n",
      "    r = make_query(query)\n",
      "NameError: name 'make_query' is not defined\n",
      "\n",
      "\n",
      "\n",
      "Absent urls: 0\n",
      "Total urls: 0\n",
      "Hits total for all urls: 0\n"
     ]
    },
    {
     "ename": "ZeroDivisionError",
     "evalue": "float division by zero",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mZeroDivisionError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-23-f50a768dc2d1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mdatasets_fpaths\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"datasets/*txt\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mgenerate_absent_datasets\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdatasets_fpaths\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/kb2vec/dataset_stats.py\u001b[0m in \u001b[0;36mgenerate_absent_datasets\u001b[0;34m(datasets_fpaths)\u001b[0m\n\u001b[1;32m     55\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Total urls:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtotal_urls\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     56\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Hits total for all urls:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtotal_hits\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 57\u001b[0;31m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Avg. hits per url: {:.2f}\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtotal_hits\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0mtotal_urls\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     58\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mZeroDivisionError\u001b[0m: float division by zero"
     ]
    }
   ],
   "source": [
    "from dataset_stats import generate_absent_datasets\n",
    "\n",
    "datasets_fpaths = \"datasets/*txt\"\n",
    "generate_absent_datasets(datasets_fpaths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DQL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "query_and_save('allUris:\"barackobama.com\"', \"data/all-uris.json\")\n",
    "query_and_save('wikipediaUri:\"en.wikipedia.org/wiki/Barack_Obama\"', \"data/wiki-uri.json\")\n",
    "query_and_save('allUris:\"en.wikipedia.org/wiki/Barack\\_Obama\"', \"data/all-uris-wiki.json\")\n",
    "query_and_save('origins:\"en.wikipedia.org/wiki/Barack_Obama\"', \"data/origins.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for entity_type in entity_types:\n",
    "    query_and_save(\n",
    "        query='type:{}'.format(entity_type),\n",
    "        output_fpath=\"data/{}.json\".format(entity_type))\n",
    "    \n",
    "query_and_save(\n",
    "    query='type:Person name:\"Alexander Panchenko\"',\n",
    "    output_fpath=\"data/ap.json\")\n",
    "\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person employments.employer.name:\"Diffbot\"',\n",
    "    output_fpath=\"data/diffbot-employees.json\")\n",
    "\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person employments.{title:\"CEO\" employer.name:\"Diffbot\"}',\n",
    "    output_fpath=\"data/diffbot-ceo.json\")\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person employments.{employer.name:\"Diffbot\" isCurrent:true}',\n",
    "    output_fpath=\"data/diffbot-current-employees.json\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Testing type of links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "query_and_save(\n",
    "    query='type:Person name:\"Angela Merkel\"',\n",
    "    output_fpath=\"data/am.json\")\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person name:\"Barack Obama\"',\n",
    "    output_fpath=\"data/bo.json\")\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person name:\"Nicolas Sarkozy\"',\n",
    "    output_fpath=\"data/ns.json\")\n",
    "\n",
    "query_and_save(\n",
    "    query='type:Person name:\"Diego Maradona\"',\n",
    "    output_fpath=\"data/dm.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
